src/widgets/programs/programs.resource.tsx

```diff
@@ -1,29 +1,55 @@
 import { openmrsObservableFetch, openmrsFetch } from "@openmrs/esm-api";
-import { Observable, of } from "rxjs";
-import { map, take } from "rxjs/operators";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { FetchResponse } from "@openmrs/esm-api/dist/openmrs-fetch";
+import { PatientProgram, Program, LocationData, SessionData } from "../types";
 
 export function fetchEnrolledPrograms(
   patientID: string
 ): Observable<PatientProgram[]> {
-  return openmrsObservableFetch(
+  return openmrsObservableFetch<FetchResponse<PatientProgram[]>>(
     `/ws/rest/v1/programenrollment?patient=${patientID}`
   ).pipe(
     map(
-      ({ data }) =>
-        data["results"].sort((a, b) =>
-          b.dateEnrolled > a.dateEnrolled ? 1 : -1
-        ),
-      take(3)
+      ({ data }: { data: FetchResponse<PatientProgram[]> }) => data["results"]
+    ),
+    map((results: PatientProgram[]) =>
+      results.sort((a, b) => (b.dateEnrolled > a.dateEnrolled ? 1 : -1))
+    )
+  );
+}
+
+export function fetchActiveEnrollments(
+  patientID: string
+): Observable<PatientProgram[]> {
+  return openmrsObservableFetch<FetchResponse<PatientProgram[]>>(
+    `/ws/rest/v1/programenrollment?patient=${patientID}`
+  ).pipe(
+    map(
+      ({ data }: { data: FetchResponse<PatientProgram[]> }) => data["results"]
+    ),
+    map((results: PatientProgram[]) =>
+      results
+        .filter(res => !res.dateCompleted)
+        .sort((a, b) => (b.dateEnrolled > a.dateEnrolled ? 1 : -1))
     )
   );
 }
 
 export function getPatientProgramByUuid(
   programUuid: string
 ): Observable<PatientProgram> {
-  return openmrsObservableFetch(
+  return openmrsObservableFetch<FetchResponse<PatientProgram>>(
```

Comment:
Should be just `PatientProgram`

Reply:
Totally right, amended in the latest commit.